java/samples/sample-code/src/main/java/com/microsoft/semantickernel/Config.java

```diff
@@ -7,56 +7,189 @@
 import com.azure.core.credential.AzureKeyCredential;
 import com.microsoft.semantickernel.util.AIProviderSettings;
 import com.microsoft.semantickernel.util.AzureOpenAISettings;
+import com.microsoft.semantickernel.util.ClientSettings;
 import com.microsoft.semantickernel.util.OpenAISettings;
 
 import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Supplier;
 
 public class Config {
 
-    public static final String CONF_PROPERTIES =
-            System.getProperty("CONF_PROPERTIES", "java/samples/conf.properties");
+    private static final List<String> DEFAULT_PROPERTIES_LOCATIONS = Arrays.asList(
+            "java/samples/conf.properties",
+            System.getProperty("user.home") + "/.sk/conf.properties"
+    );
+
+    public static final String CONF_PROPERTIES = getEnvOrProperty("CONF_PROPERTIES", "java/samples/conf.properties");
 
     public static final String OPENAI_CLIENT_TYPE =
-            System.getProperty("OPENAI_CLIENT_TYPE", "OPEN_AI");
+            getEnvOrProperty("OPENAI_CLIENT_TYPE", "OPEN_AI");
+
+    private static String getEnvOrProperty(String propertyName, String defaultValue) {
+        String env = System.getenv(propertyName);
+        if (env != null && !env.isEmpty()) {
+            return env;
+        }
+
+        String property = System.getProperty(propertyName);
+        if (property != null && !property.isEmpty()) {
+            return property;
+        }
+
+        return defaultValue;
+    }
 
+    /**
+     * Looks for settings in the order of:
+     * <p>
+     * 1. Properties file
+     * 2. System properties
+     * 3. Environment variables
+     * 4. Properties file: java/samples/conf.properties
+     * 5. Properties file: ~/.sk/conf.properties
+     *
+     * @return A client instance
+     * @throws IOException If the settings are not found
+     */
     public static OpenAIAsyncClient getClient() throws IOException {
         return ClientType.valueOf(OPENAI_CLIENT_TYPE).getClient();
     }
 
+    private static Supplier<ClientSettings<?>> getOpenaiClientFromDefaultPropertiesLocations(ClientType type) {
```

Comment:
Should be consistent in use of OpenAI is class and method names so change to `getOpenAIClientFromDefaultPropertiesLocations`